wremnants/muon_calibration.py

```diff
@@ -240,28 +241,30 @@ def define_uncrct_reco_muon_kinematics(df, kinematic_vars = ["pt", "eta", "phi",
     return df
 
 def transport_smearing_weights_to_reco(
-    resultdict, procs = ['WplusmunuPostVFP', 'WminusmunuPostVFP', 'ZmumuPostVFP'],
-    proj_axes = ['eta', 'pt', 'charge', 'passIso', 'passMT']
+    resultdict,
+    procs = ['WplusmunuPostVFP', 'WminusmunuPostVFP', 'ZmumuPostVFP'],
 ):
     for proc in procs:
-        proc_hists = resultdict[proc]['output']
-        nominal_gen_smear = (
-            proc_hists['nominal_gen_smeared'].project(*proj_axes))
-        msv_sw_gen_smear = [
-            (proc_hists['muonScaleSyst_responseWeights_gensmear'][...,0,0].project(*proj_axes)),
-            (proc_hists['muonScaleSyst_responseWeights_gensmear'][...,0,1].project(*proj_axes))
-        ]
-        sw_per_bin_gen_smear = [hh.divideHists(x, nominal_gen_smear) for x in msv_sw_gen_smear]
-        nominal_reco = proc_hists['nominal'].project(*proj_axes)
-        msv_sw_reco = [hh.multiplyHists(nominal_reco, x) for x in sw_per_bin_gen_smear]
-        proc_hists['muonScaleSyst_responseWeights'] = hh.combineUpDownVarHists(*msv_sw_reco)
+        nominal_gen_smear = resultdict[proc]['output']['nominal_gen_smeared']
```

Comment:
Probably a good idea to have some checks that the histograms exist here, I've had it happen in the past that this code is out of sync and it fails right at the end before writing the output, which is a bit annoying if you're doing other things anyway. If anything is missing you could throw from here and catch it in the main function with a warning message

Reply:
✅